lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -168,11 +168,11 @@ def commit(action, post, account = 'ws', password = @options[:password])
           return Response.new(false, 'Invalid credentials', {}, :test => test?)
         when '403'
           return Response.new(false, 'Not allowed', {}, :test => test?)
-        when '422'
-          return Response.new(false, 'Unprocessable Entity', {}, :test => test?)
-        when '500'
-          if e.response.body.split(' ')[0] == 'validation'
-            return Response.new(false, e.response.body.split(' ', 3)[2], {}, :test => test?)
+        when '422', '500'
+          if e.response.body.split(/\W+/).any? { |word| word  == 'validation' || 'configuration' || 'security' }
```

Comment:
`word == 'validation' || 'configuration' || 'security'` is always truthy, since `'configuration'` and `'security'` are truthy. I think you meant `...any? { |word| %q(validation configuration security).include?(word) }` or something similar.

Reply:
👍